src/bp/ui-studio/src/web/actions/index.ts

```diff
@@ -171,7 +171,10 @@ export const duplicateFlow: (flow: { flowNameToDuplicate: string; name: string }
 
 type AllPartialNode = (Partial<sdk.FlowNode> | Partial<sdk.TriggerNode> | Partial<sdk.ListenNode>) & Partial<FlowPoint>
 
-export const updateFlowNode: (props: AllPartialNode) => void = wrapAction(requestUpdateFlowNode, updateCurrentFlow)
+export const updateFlowNode: (props: AllPartialNode | (AllPartialNode & { id: number }[])) => void = wrapAction(
```

Comment:
`sdk.FlowNode` already contains an id key but is of type string (check the definition of union type `AllPartialNode`, line 172). Can you use this property instead of using an intersection type?

Reply:
I agree that type is wrong, it should be string, but since an id is required if using arrays and Partial makes all the properties optional, I think we should keep the intersection to avoid bugs, what do you think?

Reply:
Why not simply use `sdk.FlowNode` in this case? I personally, find it weird to have to override a Partial type to make sure some property isn't optional!

Reply:
That would make some properties required, like the name. They are other ways of making a specific property required in a partial type, but there is more code involved, I changed the type to `AllPartialNode | (AllPartialNode & Pick<Required<sdk.FlowNode>, 'id'>)[]`, let me know what you think about it. In my option, although overriding the partial type seems weird, it makes sense, because the id is required to make an array argument work.

Reply:
Okay I now understand the bigger picture. I think you can leave it like that even if it feels weird!